pkg/app/api/grpcapi/piped_api.go

```diff
@@ -963,6 +967,129 @@ func (a *PipedAPI) ReportUnregisteredApplicationConfigurations(ctx context.Conte
+			{
+				Field:    "ProjectId",
+				Operator: datastore.OperatorEqual,
+				Value:    projectID,
+			},
+		}
+
+		if matcher.Name != "" {
+			filters = append(filters, datastore.ListFilter{
+				Field:    "Name",
+				Operator: datastore.OperatorEqual,
+				Value:    matcher.Name,
+			})
+		}
+
+		// TODO: Support find node apps by appKind and appLabels.
+
+		apps, _, err := a.applicationStore.ListApplications(ctx, datastore.ListOptions{
+			Filters: filters,
+		})
+		if err != nil {
+			return nil, nil, err
+		}
+
+		nodes := make([]*model.ChainNode, 0, len(apps))
+		for _, app := range apps {
+			nodes = append(nodes, &model.ChainNode{
+				ApplicationRef: &model.ChainApplicationRef{
+					ApplicationId:   app.Id,
+					ApplicationName: app.Name,
+				},
+			})
+		}
+
+		return nodes, apps, nil
+	}
+
+	chainBlocks := make([]*model.ChainBlock, 0, len(req.Matchers)+1)
+	// Add the first deployment which created by piped as the first block of the chain.
+	chainBlocks[0] = &model.ChainBlock{
+		Index: 0,
+		Nodes: []*model.ChainNode{
+			{
+				ApplicationRef: &model.ChainApplicationRef{
+					ApplicationId:   req.FirstDeployment.ApplicationId,
+					ApplicationName: req.FirstDeployment.ApplicationName,
+				},
+				DeploymentRef: &model.ChainDeploymentRef{
+					DeploymentId: req.FirstDeployment.Id,
+				},
+			},
+		},
+	}
+
+	apps := make([]*model.Application, 0)
+	for i, filter := range req.Matchers {
+		nodes, blockApps, err := buildChainNodes(filter)
+		if err != nil {
+			return nil, err
+		}
+
+		apps = append(apps, blockApps...)
+		chainBlocks = append(chainBlocks, &model.ChainBlock{
+			Index: int32(i + 1),
+			Nodes: nodes,
+		})
+	}
+
+	dc := model.DeploymentChain{
+		Id:        uuid.New().String(),
+		ProjectId: projectID,
+		Blocks:    chainBlocks,
+	}
+
+	// Create a new deployment chain instance to control newly triggered deployment chain.
+	if err := a.deploymentChainStore.AddDeploymentChain(ctx, &dc); err != nil {
+		a.logger.Error("failed to create deployment chain", zap.Error(err))
+		return nil, status.Error(codes.Internal, "failed to trigger new deployment chain")
+	}
+
+	firstDeployment := req.FirstDeployment
+	firstDeployment.DeploymentChainId = dc.Id
+	// Trigger new deployment for the first application by store first deployment to datastore.
+	if err := a.deploymentStore.AddDeployment(ctx, firstDeployment); err != nil {
+		a.logger.Error("failed to create deployment", zap.Error(err))
+		return nil, status.Error(codes.Internal, "failed to trigger new deployment for the first application in chain")
+	}
+
+	// Make sync application command for applications of the chain.
+	for _, app := range apps {
+		cmd := model.Command{
+			Id:            uuid.New().String(),
+			PipedId:       app.PipedId,
+			ApplicationId: app.Id,
+			ProjectId:     app.ProjectId,
+			Commander:     dc.Id,
+			// TODO: Add new command type in order to separate this in chain sync command with other commands.
+			Type: model.Command_SYNC_APPLICATION,
```

Comment:
Should we use the newly created command instead of this?

Reply:
Will address with a separated PR 🙏

Reply:
Changed my mind, lets me address it in this PR

Reply:
Sure.